### PR TITLE
Support construct a graph for multi-task learning

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
One PR of #789
Also solve #651 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/transform.py [hunk 1]
```diff
@@ -1725,76 +1725,100 @@ def parse_label_ops(confs, is_node):
...
+    if len(label_ops) > 1:
+        # check whether train/val/test mask names are
+        # different for different labels
+        mask_names = []
+        for ops in label_ops:
+            mask_names.append(ops.train_mask_name)
+            mask_names.append(ops.val_mask_name)
+            mask_names.append(ops.test_mask_name)
+        assert len(mask_names) == len(set(mask_names)), \
+            f"Some train/val/test mask field names are duplicated, please check: {mask_names}."
```
Comment (jalencato) [inline python/graphstorm/gconstruct/transform.py:1819]: I think the code here will check all the masks field name for both node and edge. We should include that in the doc about that each mask name for different types of nodes/edges should be different.

File: python/graphstorm/gconstruct/transform.py [hunk 2]
```diff
@@ -1810,9 +1834,12 @@ def process_labels(data, label_processors):
...
     dict of tensors : labels (optional) and train/validation/test masks.
     """
-    print(label_processors)
-    assert len(label_processors) == 1, "We only support one label per node/edge type."
-    return label_processors[0](data)
+    assert len(label_processors) >= 1, \
+        "Number of label_processors must be one or more."
+    ret = {}
+    for label_processor in label_processors:
+        ret.update(label_processor(data))
```
Comment (jalencato) [inline python/graphstorm/gconstruct/transform.py:1841]: What about adding printing for the label_processors back here?

File: tests/end2end-tests/data_process/test_multitask_data.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+# Test the first node data
+assert g.nodes['node1'].data['feat'].dtype is th.float32
+data = g.nodes['node1'].data['feat'].numpy()
+orig_ids = np.array([reverse_node1_map[new_id] for new_id in range(g.number_of_nodes('node1'))])
+assert np.all(data <= 2)
+assert np.all(data >= -2)
+
+label_class = g.nodes['node1'].data['label_class'].numpy()
+assert "train_mask_class" in g.nodes['node1'].data
+assert "val_mask_class" in g.nodes['node1'].data
```
Comment (jalencato) [inline tests/end2end-tests/data_process/test_multitask_data.py:83]: Why not include ``test_mask_class`` here? It is included in the above:                
```
"mask_field_names": ["train_mask_class",
                      "val_mask_class",
                      "test_mask_class"],
```

**Review by jalencato — COMMENTED** [review]
A small comment for test.

File: python/graphstorm/gconstruct/transform.py (hunk 1 above)
Comment (classicsong) [reply python/graphstorm/gconstruct/transform.py:1819]: mask names for different node types can be the same. This function is called per node/edge type.

File: python/graphstorm/gconstruct/transform.py (hunk 2 above)
Comment (classicsong) [reply python/graphstorm/gconstruct/transform.py:1841]: What do you mean by printing?
Adding a debug info?

Comment (jalencato) [reply python/graphstorm/gconstruct/transform.py:1841]: Yep. I see we originally have print(label_processors)

File: tests/end2end-tests/data_process/test_multitask_data.py (hunk 3 above)
Comment (classicsong) [reply tests/end2end-tests/data_process/test_multitask_data.py:83]: Actually, we do need these two asserts. We are accessing these masks in the following tests.

File: python/graphstorm/gconstruct/transform.py (hunk 2 above)
Comment (classicsong) [reply python/graphstorm/gconstruct/transform.py]: Added a logging.debug

File: python/graphstorm/gconstruct/transform.py [hunk 4]
```diff
@@ -1725,76 +1725,100 @@ def parse_label_ops(confs, is_node):
...
-            return [CustomLabelProcessor(col_name=label_col, label_name=label_col,
-                                         id_col=(confs["source_id_col"], confs["dest_id_col"]),
-                                         task_type=task_type, train_idx=train_idx, val_idx=val_idx,
-                                         test_idx=test_idx, stats_type=label_stats_type,
-                                         mask_field_names=mask_field_names)]
-        else:
-            raise AttributeError("Custom data segmentation should be "
-                                 "applied to either node or edge tasks.")
+    assert len(label_confs) >= 1, \
+        "There must be at least one label configuration under the 'labels' config block."
```
[the remark this answers is not in the document]
Comment (classicsong) [reply python/graphstorm/gconstruct/transform.py:1729]: Updated

File: tests/end2end-tests/data_process/test_multitask_data.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+argparser.add_argument("--with-reverse-edge",
+                       type=lambda x: (str(x).lower() in ['true', '1']),
+                       default=False,
+                       help="Whether check reverse edges")
+args = argparser.parse_args()
+out_dir = args.graph_dir
+with open(args.conf_file, 'r') as f:
+    conf = json.load(f)
+
+args = argparser.parse_args()
```
[the remark this answers is not in the document]
Comment (classicsong) [reply tests/end2end-tests/data_process/test_multitask_data.py]: deleted

**Review by zhjwy9343 — APPROVED** [review]
LGTM

**Pull request merged**